rust_dev_preview/cross_service/rest_ses/src/work_item/repository.rs

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! The repository module includes data access components. It exposes Create,
+//! Retrieve, List, Update, and Delete functions that work with an `&RdsClient`
+//! to execute SQL that manages WorkItem persistence. Its interface encapsulates
+//! RDS and data parsing errors behind the WorkItemError enum defined in the
+//! parent work_item mod.
+use aws_sdk_rdsdata::{
+    error::ExecuteStatementError, model::RecordsFormatType, output::ExecuteStatementOutput,
+    types::SdkError,
+};
+use serde_json::from_str;
+
+use super::{WorkItem, WorkItemArchived, WorkItemError};
+use crate::{client::RdsClient, params};
+
+pub const RDS_DATE_FORMAT: &str = "%Y-%m-%d";
+
+/// Create a new WorkItem using `INSERT INTO`.
+/// After inserting the item, confirm its success by retrieving that item rather than the stored copy.
+#[tracing::instrument(name = "Repository Create new WorkItem", skip(item, client))]
+pub async fn create(item: WorkItem, client: &RdsClient) -> Result<WorkItem, WorkItemError> {
+    client
+        .execute_statement()
+        .sql(
+            r#"
+            INSERT INTO Work
+            (idwork, username, date, description, guide, status, archive)
+            VALUES
+            (:idwork, :username, :date, :description, :guide, :status, :archive)
+        "#,
+        )
+        .set_parameters(params![
+            ("idwork", item.idwork),
+            ("username", item.name),
+            ("date", format!("{}", item.date.format(RDS_DATE_FORMAT))),
+            ("description", item.description),
+            ("guide", item.guide),
+            ("status", item.status),
+            (
+                "archive",
+                format!("{}", u8::from(&WorkItemArchived::Active))
```

Comment:
Just curious.  It looks like you are setting the archived parameter as a string.  Did you define archived in the table as "archived TINYINT(4))"?  Where is the RDS table create statement?

Reply:
This used the resources CLI, which is tinyint(4). Postgres converts the str to int.